lib/enum/arrays.rb

```diff
@@ -1,24 +1,32 @@
 class Array
 
   def sum_of_numeric_elements
+    select { |el| el.is_a?(Numeric) }.inject(:+)
   end
 
   def product_of_numeric_elements
+    select { |el| el.is_a?(Numeric) }.inject(:*)
   end
 
   def even_numeric_elements
+    select { |el| el.is_a?(Fixnum) }.select { |num| num.even?}
   end
 
   def odd_numeric_elements
+    select { |el| el.is_a?(Fixnum) }.select { |num| num.odd? }
   end
 
   def string_elements
+    select { |el| el.is_a?(String) }
   end
 
   def lowercase_string_elements_as_uppercase
+    select { |el| el.is_a?(String) }.select { |str| str == str.downcase }.map { |word| word.capitalize }
```

Comment:
The second select isn't necessary. `capitalize` can be applied to already-capitalized words.